### PR TITLE
[FIX] a little fix for material.clone()

### PR DESCRIPTION
**Pull request by BAntDit**
Material.clone() does not work correctly in case material has custom shader
The issue discussion here: http://forum.playcanvas.com/t/looks-like-material-clone-has-some-issue/675

1. there is a little fix in material._cloneInternal method - this is enough if shader assigned to base material.

2. and there is a little fix for clone method of pc.PhongMaterial. In case custom shader assigned on phong material, this is no longer phong material, so there is no longer needed calls clone.update() for restoring default phong parameters.

### CODE REVIEW

File: src/scene/scene_material.js [hunk 1]
```diff
@@ -105,9 +105,18 @@ pc.extend(pc, function () {
...
+        
+        // in case user assign some custom shader 
+        // we need copy a shader
+        clone.shader = this.shader;
 
         clone.parameters = {};
+        
+        // and need copy parameters of that shader
+        for(var parameterName in this.parameters) {
+			if (!!this.parameters[parameterName])
```
Comment (Maksims) [inline src/scene/scene_material.js]: `tabs`? We don't do tabs :)
Use 4 spaces please.

**Comment by guycalledfrank** [comment]
Looks fine to me. Why the shader was nulled in the old version anyway?

**Comment by willeastcott** [comment]
@guycalledfrank I can't quite remember. Perhaps because I assumed it would be regenerated at render time. But yes, that doesn't account for the case where you set the shader manually (when using a pc.Material instead of a pc.PhongMaterial).

**Comment by daredevildave** [comment]
Hey, this looks good. Can you convert tabs to spaces and fill in the CLA here: https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform

Then we can merge.

**Comment by BAntDit** [comment]
Ok, I've done conversion from tabs to 4 spaces. And fill in the CLA. 
Not quite sure, does i need make new pull request after file changes? (i didn't work with github before :))

**Comment by Maksims** [comment]
It's been long, sorry, but merged now.

**Pull request merged**